crates/holochain/src/conductor/entry_def_store.rs

```diff
@@ -122,6 +124,29 @@ impl BufferedStore for EntryDefBuf {
     }
 }
 
+/// Get an [EntryDef] from the entry def store
+/// or fallback to running the zome
+pub(crate) async fn get_entry_def(
+    entry_type: &AppEntryType,
+    zome: Zome,
+    dna_file: &DnaFile,
+    conductor_api: &impl CellConductorApiT,
+) -> EntryDefStoreResult<Option<EntryDef>> {
+    let entry_def_index = u8::from(entry_type.id()) as usize;
+
+    // Try to get the entry def from the entry def store
+    let key = EntryDefBufferKey::new(zome, entry_type.id());
+    let entry_def = { conductor_api.get_entry_def(&key).await };
+
+    // If it's not found run the ribosome and get the entry defs
+    match &entry_def {
+        Some(_) => Ok(entry_def),
+        None => Ok(get_entry_defs(dna_file.clone())?
+            .get(entry_def_index)
+            .map(|(_, v)| v.clone())),
+    }
```

Comment:
Not necessarily related to this PR but can you remind me how EntryDefs get stored? It looks like `get_entry_def` looks up the EntryDefs, in the DnaStore, but then there is also a separate `EntryDefBuf`. How do those relate? I noticed this because the comment says it gets defs from the "entry def store".

Reply:
get_entry_def checks the EntryDefBuf for the EntryDef but then falls back to getting it from the wasm.
The DnaStore has the wasms.
The EntryDefStore has the EntryDefs and the EntryDefBuf is an in memory wrapper around it.
Basically the EntryDef should be loaded in memory but if it isn't then we fallback to getting it from the wasm.